crates/sp-domains/src/merkle_tree.rs

```diff
@@ -45,7 +45,7 @@ impl Hasher for Blake2b256Algorithm {
     fn hash(data: &[u8]) -> Blake3Hash {
         let mut hasher = Blake2b::new();
         hasher.update(data);
-        hasher.finalize_fixed().into()
+        Blake3Hash::from_array_ref(hasher.finalize_fixed().as_ref())
     }
```

Comment:
I think the existing unmodified code above is wrong, and that's the cause of some of the trickiest parts of this PR:
```rust
impl Hasher for Blake2b256Algorithm {
    type Hash = Blake3Hash;
    ...
}
```

@nazar-pc what should the Blake2 hash type be here?

Reply:
Yeah, `Blake3Hash` only compiled by accident before. We have switched from Blake2 to Blake3 some time ago, which is probably why we have it here. I think this should return `[u8; 32]` or something like that instead. Unless we can switch to Blake3 here, which might be nice.

Reply:
I think if we want to change the hash algorithm, that should be a separate PR from a refactor.

Reply:
I'm not sure we want to change the algorithm at all :thinking: